recipes/haproxy_osuosl.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Cookbook:: osl-keepalived
+# Recipe:: haproxy-osuosl
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include_recipe 'osl-keepalived::default'
+
+haproxy_osuosl = node['osl-keepalived']['haproxy-osuosl']
+secrets = data_bag_item('osl_keepalived', 'haproxy_osuosl')
+
+if node['fqdn'] == haproxy_osuosl['default_master']
+  master = true
+  priority = 200
+else
+  master = false
+  priority = 100
+end
```

Comment:
I'm wondering if this could be done differently using attributes instead. Perhaps something like this (maybe refactoring what I mentioned above):

``` ruby
node['osl-keepalived']['master'] = {
  'lb1.osuosl.org' => true,
  'lb2.osuosl.org' => false
}
node['osl-keepalived']['priority'] = {
  'lb1.osuosl.org' => 200,
  'lb2.osuosl.org' => 100
}
```
i.e. Use the FQDN as the deciding factor on how it's set. Then you can just do the following in the recipe: 

``` ruby
keepalived_vrrp_instance 'haproxy-osuosl-ipv4' do
  master node['osl-keepalived']['master'][node['fqdn']]
  # ...
end
```